packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -459,6 +460,22 @@ LightningElement.prototype = {
         return getBoundingClientRect(elm);
     },
 
+    attachInternals(): ElementInternals {
+        const vm = getAssociatedVM(this);
+        const {
+            elm,
+            renderer: { attachInternals },
+        } = vm;
+
+        if (vm.renderMode === RenderMode.Light || vm.shadowMode === ShadowMode.Synthetic) {
+            throw new ReferenceError(
```

Comment:
```suggestion
            throw new Error(
```

[MDN defines a ReferenceError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError) as "an error when a variable that doesn't exist (or hasn't yet been initialized) in the current scope is referenced."

I think sometimes we try to use a ReferenceError, TypeError, etc., but I usually just use Error personally.